File: 0122.买卖股票的最佳时机II/0122-买卖股票的最佳时机II.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        if prices.len() < 2{
            return 0;
        }
        let mut ret = 0;
        for i in 0..prices.len()-1 {
            ret += 0.max( prices[i+1]-prices[i]);
        }
        
        ret
    }
}

File: 0122.买卖股票的最佳时机II/0122-买卖股票的最佳时机II.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        def window(arr, n=2):
            pre = arr[0]
            for a in arr[1:]:
                yield pre, a
                pre = a

        return sum(max(cur - pre, 0) for pre, cur in window(prices)) if prices else 0
